packages/next/src/server/lib/chrome-devtools-workspace.ts

```diff
@@ -0,0 +1,90 @@
+import type { ServerResponse } from 'http'
+import type { NextConfigComplete } from '../config-shared'
+import type { NextUrlWithParsedQuery } from '../request-meta'
+
+import { randomUUID } from 'crypto'
+import * as fs from 'fs'
+import * as path from 'path'
+import { getStorageDirectory } from '../cache-dir'
+
+// Keep the uuid in memory as it should never change during the lifetime of the server.
+let workspaceUUID: string | null = null
+
+/**
+ * https://developer.chrome.com/docs/devtools/workspaces#generate-json
+ */
+interface ChromeDevtoolsWorkspace {
+  workspace: {
+    uuid: string
+    root: string
+  }
+}
+
+export function isChromeDevtoolsWorkspaceUrl(
+  url: NextUrlWithParsedQuery
+): boolean {
+  return url.pathname === '/.well-known/appspecific/com.chrome.devtools.json'
+}
+
+export async function handleChromeDevtoolsWorkspaceRequest(
+  response: ServerResponse,
+  opts: { dir: string },
+  config: NextConfigComplete
+) {
+  response.setHeader('Content-Type', 'application/json')
+  response.end(
+    JSON.stringify(
+      await getChromeDevtoolsWorkspace(opts.dir, config.distDir),
+      null,
+      2
+    )
+  )
+}
+
+/**
+ * For https://developer.chrome.com/docs/devtools/workspaces
+ */
+export async function getChromeDevtoolsWorkspace(
+  root: string,
+  configDistDir: string
+): Promise<ChromeDevtoolsWorkspace> {
+  if (workspaceUUID === null) {
+    const distDir = path.join(root, configDistDir)
+    const cacheBaseDir = getStorageDirectory(distDir)
+
+    if (cacheBaseDir === undefined) {
+      workspaceUUID = randomUUID()
+    } else {
```

Comment:
@huozhi Do you remember why we don't want to write to a local cache in a Docker container? I see `getStorageDirectory` used in other places as well but it's not clear why we wouldn't write in a Docker container.

Reply:
That api was for persisting the data in `.next`. In Docker each build can start from the beginning, so the `.next` folder might be empty, unlike if you build on a same machine.